### DISCUSSION TITLE
@types/react-data-grid for version 7

### DISCUSSION THREAD

**Discussion by wss29** [Comcast/react-data-grid#3252] [Q&A]
Hi,
    I want to use react-data-grid in my typescript project, so I download @types/react-data-grid, but I find it only supports react-data-grid version 4
![image](https://github.com/adazzle/react-data-grid/assets/4537809/220a42c3-a460-435d-8b06-5ee9fa7611b1)
So it is hard to use grid version greater than 4. Is it exist a newer @types version for react-data-grid?

**Comment by paustint** [comment]
@wss29 - types are included in `react-data-grid` and you should not install from `@types/react-data-grid` as those are very old and no longer required.

**Reply by wss29** [reply]
Hi,
   thanks for your reply,  I install it successfully,  but there is still some problem, for example, I want to use renderSummaryCell and textEditorClassname as your CommonFeatures.tsx do, but I can not import it. my install package is as follow
![image](https://github.com/adazzle/react-data-grid/assets/4537809/66b6dc5f-1f58-4ea9-9eac-4838bc8595e6)
renderSummaryCell, textEditorClassname, and some other types like Direction do not exist in index.d.ts, how can import them,

**Comment by paustint** [comment]
Ahh - sorry, I forgot to mention that you should be using version 7, which has been in alpha+beta for 4 years....

Most of the docs and examples are based on version 7 (note: the readme is a bit shy on docs, so your best bet is to look in the website/examples folder for examples)

![image](https://github.com/adazzle/react-data-grid/assets/5461649/969cf3ab-5bf9-4423-8eba-537e1cbb1d5e)

**Reply by wss29** [reply]
Hi
   I indeed use v7
![image](https://github.com/adazzle/react-data-grid/assets/4537809/b4e8334a-9e0f-4d5f-bbce-94da96cf7293)
![image](https://github.com/adazzle/react-data-grid/assets/4537809/daaf2af1-f9fd-42f5-b99e-8f6d280d9021)
![image](https://github.com/adazzle/react-data-grid/assets/4537809/1536607b-a9d2-4299-b687-2b2e40835ede)
as you can see renderEditCell and renderSummaryCell do not exist in index.d.ts , and I find some other types in your website/demo also do not exist in index.d.ts
